lib/adapters/REST/endpoints/upload.ts

```diff
@@ -1,13 +1,23 @@
 import type { AxiosInstance } from 'contentful-sdk-core'
 import { Stream } from 'stream'
-import { GetSpaceParams } from '../../../common-types'
+import { GetSpaceEnvironmentUploadParams } from '../../../common-types'
 import { getUploadHttpClient } from '../../../upload-http-client'
 import { RestEndpoint } from '../types'
 import * as raw from './raw'
 
+const getUploadUrlPath = (params: GetSpaceEnvironmentUploadParams) => {
+  const spacePath = `/spaces/${params.spaceId}/uploads`
+  const environmentPath = `/spaces/${params.spaceId}/environments/${params.environmentId}/uploads`
+  const path = params.environmentId ? environmentPath : spacePath
+
+  // Return path as is if uploadId is not provided, otherwise append it to the path
+  if (!params.uploadId) return path
```

Comment:
Hmm... I don't think so. The path for the post route would look like `/spaces/${params.spaceId}/uploads` OR like `/spaces/${params.spaceId}/environments/${params.environmentId}/uploads`. So no asset ids, or upload ids in the route being added here. This is basically just a refactoring of the path logic since the post doesn't need an upload id but get and deletes do. 

The asset itself is passed as a function parameter like this
```
  const path = getUploadUrlPath(params)
  return raw.post(httpUpload, path, file, {
    headers: {
      'Content-Type': 'application/octet-stream',
    },
  })
}
```

Reply:
👍🏻 But then we shouldn't add the `uploadId` to the path here, right?

Reply:
Correct, `const path = getUploadUrlPath(params)` will ensure that the path will contain the upload id for get/delete but not for the create operation.

Reply:
Got it. 👍🏻 A bit of a nit but I think I would prefer to have this reflected in the types of the `create` [method](https://github.com/contentful/contentful-management.js/blob/018840f2b6d07ad1f312964beb82319f889e0dc1/lib/adapters/REST/endpoints/upload.ts#L20C11-L20C42) then, basically the `create` method then should not have `GetSpaceEnvironmentUploadParams` as type, because it will never accept the optional `uploadId` (or if it did, it would result in a 400 api call), but only `GetSpaceEnvironmentParams` right?

Reply:
Done, I agree, the create method should take in only the GetSpaceEnvironmentParams

Reply:
Yeah I have a few cents to add here as well... I think the comments and the suggestions on the approach make a lot of sense.

My opinion is that we should not overload the method here to handle both uses cases but treat them as different things. This also makes the type definition itself a bit cleaner.

So to give you an example picked at random from the same folder we're in here (REST/endpoints): https://github.com/contentful/contentful-management.js/blob/cebe70ac8963204e8341506eb6c93e1e4d1874d3/lib/adapters/REST/endpoints/organization-membership.ts#L15-L19

In general this is true of every "restful" resource, there will always be two endpoints the "single" (for read/update/delete) and the "collection" (for create/query). So I suggest we embrace it, create two methods which have different input types, and then update `GetSpaceEnvironmentUploadParams` type to require the uploadid (so it can be reused throughout).

I think with this small change you might even end up with less lines of code, certainly one less conditional, and hopefully a cleaner solution.

Pretty small thing at the end of the day but i think worth getting to a nice clean state.